File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/request/AddCartItemRequest.java
package com.coffee.gu.api.controller.v1.request;

import com.coffee.gu.CoreException;
import com.coffee.gu.ErrorType;
import com.coffee.gu.cart.AddCartItem;


public record AddCartItemRequest(
        Long menuId,
        Long quantity
) {
    public AddCartItem toAddCartItem() {
        if (quantity <= 0) throw new CoreException(ErrorType.INVALID_REQUEST, null);
        return new AddCartItem(menuId, quantity);
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/request/AddCartItemRequest.kt
package com.coffee.gu.api.controller.v1.request

import com.coffee.gu.CoreException
import com.coffee.gu.ErrorType
import com.coffee.gu.cart.AddCartItem

class AddCartItemRequest(
    val menuId: Long,
    val quantity: Long,
) {
    fun toAddCartItem(): AddCartItem {
        if (quantity <= 0) throw CoreException(ErrorType.INVALID_REQUEST, null)
        return AddCartItem(menuId, quantity)
    }
}
